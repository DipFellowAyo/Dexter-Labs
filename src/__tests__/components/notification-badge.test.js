import React from 'react'
import Badge from '@material-ui/core/Badge'
import Box from '@material-ui/core/Box'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

import NotificationBadge from '../../components/notification-badge'

describe('Notification Badge Component', () => {
  let notificationBadge

  beforeEach(() => {
    notificationBadge = shallow(<NotificationBadge />)
  })
  afterAll(() => jest.resetAllMocks())

  it('should render without crashing', () => {
    expect(notificationBadge).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(toJson(notificationBadge)).toMatchSnapshot()
  })

  it('should render the notification badge with a variant props', () => {
    expect(notificationBadge.find(Box)).toBeTruthy()
  })

  it('should render the Box with a component props', () => {
    notificationBadge.setProps({ component: 'img' })
    expect(notificationBadge.find(Box).props().component).toBe('img')
  })

  it('should render the Box with a height props', () => {
    notificationBadge.setProps({ alt: 'notification' })
    expect(notificationBadge.find(Box).props().alt).toBe('notification')
  })

  it('should render the Box with a width props', () => {
    notificationBadge.setProps({ src: '/vectors/notification.svg' })
    expect(notificationBadge.find(Box).props().src).toBe(
      '/vectors/notification.svg'
    )
  })
})
