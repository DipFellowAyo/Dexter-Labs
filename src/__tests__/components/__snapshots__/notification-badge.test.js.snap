// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Badge Component should match the snapshot 1`] = `
<WithStyles(ForwardRef(IconButton))
  aria-label="notification"
  className="makeStyles-iconButtonStyle-1"
>
  <WithStyles(WithStyles(ForwardRef(Badge)))
    badgeContent={3}
    color="secondary"
  >
    <Styled(MuiBox)
      alt="notification"
      component="img"
      src="/vectors/notification.svg"
    />
  </WithStyles(WithStyles(ForwardRef(Badge)))>
</WithStyles(ForwardRef(IconButton))>
`;
